src/quo2/components/wallet/transaction_progress/view.cljs

```diff
@@ -0,0 +1,306 @@
+                                                             colors/neutral-60)]
+    (or (= network-state :confirmed)
+        (= network-state :finalising)) ["positive-state"
+                                        (colors/theme-colors colors/white
+                                                             colors/success-50
+                                                             colors/success-60)]
+    (= network-state :finalized)       ["diamond"
+                                        (colors/theme-colors colors/white
+                                                             colors/success-50
+                                                             colors/success-60)]
+    (= network-state :error)           ["negative-state"
+                                        (colors/theme-colors colors/danger
+                                                             colors/danger-50
+                                                             colors/danger-60)]))
+
+(defn transaction-progress
+  [{:keys [title
+           on-press
+           accessibility-label
+           network-type
+           network-state
+           start-interval-now
+           override-theme]}]
+  (let [count (reagent/atom 0)]
+    (rn/use-effect
+     (fn []
+       (when start-interval-now
+         (start-interval network-state))
+       (clear-counter)
+       (fn []
+         (stop-interval)))
+     [network-state])
+    [rn/view
+     [rn/touchable-without-feedback
+      {:on-press            on-press
+       :accessibility-label accessibility-label}
+      [rn/view
+       {:style style/box-style}
+       [rn/view
+        {:style style/title-item-container}
+        [rn/view
+         {:style style/inner-container}
+         [load-icon "placeholder" (colors/theme-colors colors/white colors/neutral-50 colors/neutral-60)]
+         [rn/view
+          {:style style/title-container}
+          [render-text title override-theme]]
+         (if (= network-state :error)
+           [button/button
+            {:size      24
+             :icon-left :i/refresh
+             :type      :primary} "Retry"])]]
+       [rn/view
+        {:style style/padding-row}
+        [quo2/context-tag {:blur? [false]}
+         (resources/get-mock-image :collectible)
+         "Doodle #120"]]
+       (if (= network-type :mainnet)
+         [rn/view
+          {:style style/item-container}
+          [rn/view
+           {:style (assoc style/progress-container
+                          :border-color
+                          (colors/theme-colors colors/white colors/neutral-10 colors/neutral-80))}
+           (let [[status-icon color] (get-status-icon network-type network-state)]
+             [load-icon status-icon color])
+           [rn/view
+            {:style style/title-container}
+            [render-text (str (network-type-text network-type network-state) " Mainnet") override-theme
+             :typography
```

Comment:
`render-text`'s signature is more complicated than it needs to be. You could make it accept a single map instead of two positional args and the extra variadic args.

```clojure
[render-text
 {:title      (str (network-type-text network-type network-state) " Mainnet")
  :theme      override-theme
  :typography :typography/font-regular
  :weight     :regular
  :size       :paragraph-2}]
```